examples/subscriber_multithreaded.rs

```diff
@@ -2,39 +2,25 @@
 // SPDX-FileCopyrightText: © Contributors to the iceoryx-rs project
 // SPDX-FileContributor: Mathias Kraus
 
-use iceoryx_rs::sb::{SampleReceiverWaitState, SubscribeState, TopicBuilder};
+use iceoryx_rs::sb::{SampleReceiverWaitState, SubscriberBuilder};
 use iceoryx_rs::Runtime;
 
 use std::error::Error;
 use std::thread;
 use std::time::Duration;
 
 #[repr(C)]
-struct CounterTopic {
+struct Counter {
     counter: u32,
 }
 
 fn main() -> Result<(), Box<dyn Error>> {
     Runtime::init("subscriber_multithreaded");
 
-    let topic = TopicBuilder::<CounterTopic>::new("Radar", "FrontLeft", "Counter")
-        .queue_capacity(5)
-        .build();
-
-    let (subscriber, sample_receive_token) = topic.subscribe_mt();
-
-    let mut has_printed_waiting_for_subscription = false;
-    while subscriber.subscription_state() != SubscribeState::Subscribed {
-        if !has_printed_waiting_for_subscription {
-            println!("waiting for subscription ...");
-            has_printed_waiting_for_subscription = true;
-        }
-        thread::sleep(Duration::from_millis(10));
-    }
-
-    if has_printed_waiting_for_subscription {
-        println!("  -> subscribed");
-    }
+    let (subscriber, sample_receive_token) =
+        SubscriberBuilder::<Counter>::new("Radar", "FrontLeft", "Counter")
+            .queue_capacity(5)
+            .create_mt()?;
```

Comment:
What does the suffix `_mt` mean and why does the publisher does not have such a suffix? When it has something todo with multithreaded wouldn't be `create_threadsafe` somehow more fitting? 

Or does the subscriber come with multiple threads?

Reply:
`_mt` means that a subscribed `Subscriber` will be created which can receive samples in a different thread than the one which created the subscriber. Internally the pointer to the C++ subscriber is stored in a smart pointer similar to `shared_ptr`. In Rust there are two types `Rc` and `Arc`. The latter has an atomic ref counting and the former not. This means the latter can be shared across threads and the latter not. Since one only pays for what one uses, there are two differrnt types for a subscriber, `st::Subscriber` and `mt::Subscriber`.